src/config/gro.config.default.ts

```diff
@@ -20,7 +20,7 @@ const createConfig: GroConfigCreator = async () => {
 				name: 'node',
 				platform: 'node',
 				input: [
-					(await pathExists('src/server/server.ts')) ? 'server/server.ts' : null!,
+					(await hasGroServer()) ? 'server/server.ts' : null!,
```

Comment:
Oooh, this is a nice addition!

Reply:
It's not documented yet, but it seems like we're now saying we have a "Gro server" and a "deprecated Gro frontend". I wonder what other "default project forms" we should support? The config is very open ended, so people can do whatever, but it's nice to have the tool work with zero config on a range of use cases.

Reply:
well and now it's not just "why the f is it looking for this file?" it's "this method checks for a gro config" so you get context on the magic string.

Reply:
Yep I added that as a checkbox in the original PR draft checklist! Programmatic linkage of these strings.

Reply:
I realize now this code may look bad out of context. The reason it was written with strings like that hardcoded in, is because the original requirements were very simple, just `src/index.html` and `src/index.ts`. More specifically though, this file is the "default" config for projects that don't define one, so the intent was to have something that should be simple and clean to copy into your project. In an older version of Gro, it copied this file into your project automatically as `src/gro.config.ts`.

So it started as this user-facing config file, but it's grown into something that needs real maintainence. I refactored it to use the Gro helpers - maybe obtuse, but correct. I'd rather err on the side of correctness over approachability, now that we have more complex requirements, like with the `server/server.ts` file. Showing people some Gro internals isn't so bad if it gets them to do good practices.

Reply:
This is related to "task pollution" mentioned here: https://github.com/feltcoop/gro/pull/128 in the `dev` task, as well as the `version` task.